proseco/guide.py

```diff
@@ -340,6 +349,45 @@ def get_initial_guide_candidates(self):
         return cand_guides
 
 
+
+def check_fid_trap(cand_stars, fids, dither):
```

Comment:
It looks like there is precedent, but I don't see why you change from `cand_guides` to `cand_stars` for the arg name.  `cand_guides` is the name of the thing you use when calling, so when reading this code I got confused, wondering if there is a different concept of candidate *stars*.  It helps code clarity (for your readers) to be consistent with naming.

Reply:
And at some level these should all just become methods of `GuideTable` and then you need no args at all.  Not a requirement for 4.0.

Reply:
I started there too, but eventually in many cases the benefit of using class methods became obvious.  For instance you could call `self.reject` (or whatever that is called) or `self.log` and thus clean up the API alot by not having those `rej` return args.

FYI the log machinery was designed to allow capture of anything in the `data` kwarg (e.g. those reject dicts), not just text messages.  Then you automatically get the function name and time where a rejection happened (FWIW).  You can toss in extra kwargs like `reject=True`.  This is how warnings are handled.

In practice you aren't doing independent unit tests now (for this particular function), and it is easy enough to do so by just making an empty `GuideTable()` and setting attributes.  See e.g.
https://github.com/sot/proseco/blob/8db4fa243b68d5744f3347b8ce08891e9c127372/proseco/tests/test_acq.py#L31

Reply:
OK.  I'll plan to clean up `cand_stars` at some point  (I think they were all `cand_stars` when I started but you liked `cand_guides` better and then I just didn't implement it universally) and move the reject crap into logging, but I think this is functional at this point.